### PR TITLE
Fix a bug on curvilinear grid generation using spherical coordinate 

### PR DESCRIPTION
**Pull request by lucacarniato**

### CODE REVIEW

**Comment by ahmad-el-sayed** [comment]
@lucacarniato could you please fix the clang format check?

File: libs/MeshKernel/src/Mesh2D.cpp [hunk 1]
```diff
@@ -37,6 +37,8 @@
 
 using meshkernel::Mesh2D;
```
Comment (ahmad-el-sayed) [inline libs/MeshKernel/src/Mesh2D.cpp:38]: I'd simply put the class def in `namespace meshkernel { .... }`.
This applies to several classes throughout the repo.

File: libs/MeshKernel/include/MeshKernel/Mesh.hpp [hunk 2]
```diff
@@ -122,6 +122,10 @@ namespace meshkernel
         /// @brief Default constructor
         Mesh() = default;
 
+        /// @brief  Constructs an empty mesh, sets only the projection
+        /// @param[in] projection  The projection to use
+        Mesh(Projection projection) : m_projection(projection){}
```
Comment (ahmad-el-sayed) [inline libs/MeshKernel/include/MeshKernel/Mesh.hpp:127]: Why not simply have a public method to set the projection in the base class Mesh? Something like
```
 Mesh::SetProjection(Projection projection) {  m_projection = projection; }
```
This saves you defining custom constructor in Mesh and all its derived classes: Mesh1D, Mesh2D and CurvilinearGrid.
Do you think this is possible?

Comment (lucacarniato) [reply libs/MeshKernel/include/MeshKernel/Mesh.hpp]: i think is better to set the projection in the constructor even if this requires a bit more code

File: libs/MeshKernel/src/Mesh2D.cpp (hunk 1 above)
Comment (lucacarniato) [reply libs/MeshKernel/src/Mesh2D.cpp:38]: I sis not use this form to avoid an extra level of nesting. But it could indeed be used

**Pull request merged**